lib/protest/runner.rb

```diff
@@ -42,6 +42,10 @@ def report(test)
     def fire_event(event, *args)
       event_handler_method = :"on_#{event}"
       @report.send(event_handler_method, *args) if @report.respond_to?(event_handler_method)
+
+      if Protest.fail_fast? && [:failure, :error].include?(event)
```

Comment:
Yes, that seemed like the natural place to exit for me, but it would imply adding an `exit 1 if Protest.fail_fast?` line below the rescue of both `AssertionFailed` and `Exception`, so I chose `fire_event` in order to avoid duplication.

Do you prefer moving the logic there and adding both lines?